cmd/maya-exporter/app/collector/zvol/pool_synctime.go

```diff
@@ -0,0 +1,137 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zvol
+
+import (
+	"os"
+	"strings"
+	"sync"
+
+	col "github.com/openebs/maya/cmd/maya-exporter/app/collector"
+	types "github.com/openebs/maya/pkg/exec"
+	zvol "github.com/openebs/maya/pkg/zvol/v1alpha1"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+// poolMetrics implements prometheus.Collector interface
+type poolMetrics struct {
+	sync.Mutex
+	*poolSyncMetrics
+	request bool
+	runner  types.Runner
+}
+
+// NewPoolSyncMetric returns new instance of poolMetrics
+func NewPoolSyncMetric(runner types.Runner) col.Collector {
+	return &poolMetrics{
+		poolSyncMetrics: newPoolMetrics().
+			withZpoolLastSyncTime().
+			withZpoolStateUnknown().
+			withzpoolLastSyncTimeCommandError(),
+		runner: runner,
+	}
+}
+
+func (p *poolMetrics) setRequestToFalse() {
+	p.Lock()
+	p.request = false
+	p.Unlock()
+}
+
+// collectors returns the list of the collectors
+func (p *poolMetrics) collectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		p.zpoolLastSyncTime,
+		p.zpoolStateUnknown,
+		p.zpoolLastSyncTimeCommandError,
+	}
+}
+
+// Describe is implementation of Describe method of prometheus.Collector
+// interface.
+func (p *poolMetrics) Describe(ch chan<- *prometheus.Desc) {
+	for _, col := range p.collectors() {
+		col.Describe(ch)
+	}
+}
+
+func (p *poolMetrics) checkError(stdout []byte) *poolfields {
+
+	if zvol.IsNoDataSetAvailable(string(stdout)) || IsNoPoolAvailable(string(stdout)) {
+		pool := poolfields{
+			name:                          os.Getenv("HOSTNAME"),
+			zpoolLastSyncTime:             0,
+			zpoolStateUnknown:             1,
+			zpoolLastSyncTimeCommandError: 0,
+		}
+		return &pool
+	}
+	return nil
+}
+
+func (p *poolMetrics) get(ch chan<- prometheus.Metric) *poolfields {
+	p.Lock()
+	defer p.Unlock()
+
+	klog.V(2).Info("Run zfs get openebs.io:livenesstimestamp")
+	stdout, err := zvol.Run(p.runner)
+	if err != nil {
+		pool := poolfields{
+			name:                          os.Getenv("HOSTNAME"),
+			zpoolLastSyncTime:             0,
+			zpoolStateUnknown:             0,
+			zpoolLastSyncTimeCommandError: 1,
+		}
+		return &pool
+	}
+
+	if pool := p.checkError(stdout); pool != nil {
+		return pool
+	}
+	klog.V(2).Infof("Parse stdout of zfs get openebs.io:livenesstimestamp command, got stdout: \n%v", string(stdout))
+	pool := poolMetricParser(stdout, p.poolSyncMetrics)
+
+	return pool
+}
+
+// Collect is implementation of prometheus's prometheus.Collector interface
+func (p *poolMetrics) Collect(ch chan<- prometheus.Metric) {
+	p.Lock()
+	p.request = true
+	p.Unlock()
+
+	pool := p.get(ch)
```

Comment:
we need to add nil check here, since you are returning nil when len(f) < 2 and add a error counter for such cases.